### PR TITLE
docs: Apply editor configurations from a ConfigMap

### PR DESCRIPTION
**Pull request by RomanNikitenko**
## What does this pull request change?
Adds a page that contains steps how to apply editor configurations using a ConfigMap.

image

image

image

## What issues does this pull request fix or reference?
It was done for the https://issues.redhat.com/browse/CRW-7258.
PR in the Che-Code: https://github.com/che-incubator/che-code/pull/460

## Pull Request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

**Review by ibuziuk — APPROVED** [review]

File: modules/administration-guide/pages/editor-configurations-for-microsoft-visual-studio-code.adoc [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+You can configure Visual Studio Code - Open Source editor by adding configurations to a ConfigMap.
+These configurations are applied to any workspace you open.
+Once a workspace is started, the editor checks for this ConfigMap and stores configurations to the corresponding config files.
+
+The following sections are currently supported:
+
+* settings.json
+* extensions.json
+
+`settings.json` section contains various settings that allow to customize different parts of the Code - OSS editor. +
```
Comment (deerskindoll) [inline modules/administration-guide/pages/editor-configurations-for-microsoft-visual-studio-code.adoc]:
```suggestion
The `settings.json` section contains various settings that allow to customize different parts of the Code - OSS editor. +
```

File: modules/administration-guide/pages/editor-configurations-for-microsoft-visual-studio-code.adoc [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+`extensions.json` section contains recommented extensions are installed when a workspace is started.
```
Comment (deerskindoll) [inline modules/administration-guide/pages/editor-configurations-for-microsoft-visual-studio-code.adoc]:
```suggestion
The `extensions.json` section contains recommented extensions are installed when a workspace is started.
```

File: modules/administration-guide/pages/editor-configurations-for-microsoft-visual-studio-code.adoc [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+.Procedure
+
+* Add a new ConfigMap to the user's {orch-namespace}, define the __settings.json__  and __extensions.json__ sections, specify the settings you want to add and the IDs of the extensions you want to install.
```
Comment (deerskindoll) [inline modules/administration-guide/pages/editor-configurations-for-microsoft-visual-studio-code.adoc]:
```suggestion
* Add a new ConfigMap to the user's {orch-namespace}, define the `__settings.json__`  and `__extensions.json__` sections, specify the settings you want to add, and the IDs of the extensions you want to install.
```

File: modules/administration-guide/pages/editor-configurations-for-microsoft-visual-studio-code.adoc [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    }
+immutable: false
+----
+====
+
+* Start or restart your workspace 
+
+[NOTE]
+====
+Ensure the Configmap contains data in a valid JSON format.
```
Comment (deerskindoll) [inline modules/administration-guide/pages/editor-configurations-for-microsoft-visual-studio-code.adoc]:
```suggestion
Make sure that the Configmap contains data in a valid JSON format.
```

**Review by deerskindoll — COMMENTED** [review]
a couple of tweaks.

I applied monospace to __settings.json__ and __extensions.json__ but I'm wondering if that was correct. what does the __ stand for?

**Comment by RomanNikitenko** [comment]
@deerskindoll 
> what does the __ stand for?

Using this way I selected what sections are expected in the ConfigMap
I don't mind to remove it, if you think it looks better without that formatting, just let me know

I think we should remove it because the names of the sections in ConfigMap don't contain the character. The monospace will serve as highlight.

**Review by deerskindoll — APPROVED** [review]
lgtm

**Comment by RomanNikitenko** [comment]
@deerskindoll 
I applied your suggestion: https://github.com/eclipse-che/che-docs/pull/2841/commits/d1b0c1362a0cdd5a920ca8a42e388cd33fce03be

thank you for your review!

**Pull request merged**